### PR TITLE
Feature/design public interface

### PR DESCRIPTION
**Pull request by andrewtran1995**
Prototyped the public interface for the ImP (Image Processing) component.

### CODE REVIEW

**Comment by andrewtran1995** [comment]
@elswenson , @agortega, @athielk, could I have at least two of you review this for me? 😄

File: imageProcessing/imageProcessingInterface.py [hunk 1]
```diff
@@ -1,6 +1,45 @@
...
+        pass
+
+    def __find_fiducial_markers(self):
+        """
+        Identify fiducial markers in __image_mat
+        :return:
+        """
+        pass
+
+    def __identify_aerocubes(self, fiducial_markers):
```
Comment (agortega) [inline imageProcessing/imageProcessingInterface.py]: how would the person get the markers?

**Comment by joyoyoyoyoyo** [comment]
So I think as an opening interface right now, this works. You wrote out a good skeleton both for ImP and for tests and the interface is in PEP8 :smile:

I think we might be able to consider the case where unordered behavior is performed.
So if someone calls image processor, this is the expected behavior

```
ImageProcessor imp = ImageProcessor()
imp.loadImage(image)
markers = imp.findMarkers() # should return markers
imp.identifyAerocubes(markers)
```

But, this is a breaking case

```
ImageProcessor imp = ImageProcessor()
imp.identifyAerocubes(markers) # This function should not be called unless it is in a state to operate.
markers = imp.findMarkers() #  find markers should only work with a loaded image in it
imp.loadImage(image)
```

I see that we can approach it in three ways:
1. The functions can be dependent on what they return.
2. We can also think of the ImageProcessor as only being able to be constructed if it is a state to execute the function. 
3. An alternative approach is to throw exceptions if we execute those commands out of order.
4. Or we have them private methods like here.

**Comment by joyoyoyoyoyo** [comment]
You got my :+1: for the Pull Request

**Pull request merged**